src/component/form/TextInputElement/TextInputElement.tsx

```diff
@@ -418,10 +420,19 @@ export class TextInputElement extends React.Component<TextInputProps, TextInputS
         });
     }
 
+    private hasErrors(): boolean {
+        return (
+            (this.state.errors && this.state.errors.length > 0) ||
+            (this.props.entry &&
+                this.props.entry.validationFailures &&
+                this.props.entry.validationFailures.length > 0)
+        );
+    }
```

Comment:
a lot of methods and checks are around to support old-world and new-world at the same time.. this will become cleaner